cpp/include/cuspatial_test/column_factories.hpp

```diff
@@ -0,0 +1,285 @@
+ *
+ * A multilinestring column has cudf type LIST<LIST<LIST<FLOAT | DOUBLE>>>
+ *
+ * Example:
+ * [
+ *    MULTILINESTRING (LINESTRING (0 0, 1 1), LINESTRING (2 2, 3 3)),
+ *    MULTILINESTRING (LINESTRING (4 4, 5 5))
+ * ]
+ * Offset 0 2 3
+ * Offset 0 2 4 6
+ * Offset 0 2 4 6 8 10
+ * Child  0 0 1 1 2 2 3 3 4 4 5 5
+ *
+ * @tparam T Coordinate value type
+ * @param multilinestring_offsets Offset to the starting position for each multilinestring
+ * @param linestring_offsets Offset to the starting position for each linestring
+ * @param point_coords Interleaved x-y coordinates of the points
+ * @param stream The CUDA stream on which to perform computations
+ *
+ * @return A cudf LIST column with multilinestring data
+ */
+template <typename T>
+std::pair<collection_type_id, std::unique_ptr<cudf::column>> make_linestring_column(
+  std::initializer_list<cudf::size_type>&& multilinestring_offsets,
+  std::initializer_list<cudf::size_type>&& linestring_offsets,
+  std::initializer_list<T> linestring_coords,
+  rmm::cuda_stream_view stream)
+{
+  auto num_points = linestring_coords.size() / 2;
+  return {
+    collection_type_id::MULTI,
+    make_non_nullable_lists_column<T>(
+      multilinestring_offsets,
+      make_non_nullable_lists_column<T>(
+        linestring_offsets,
+        make_non_nullable_lists_column<T>(coords_offsets(num_points, stream), linestring_coords)))};
+}
+
+/**
+ * @brief helper function to make a polygon column
+ *
+ * A multilinestring column has cudf type LIST<LIST<LIST<FLOAT | DOUBLE>>>
+ *
+ * Example:
+ * [
+ *    POLYGON ((0 0, 1 1, 0 1, 0 0), (0 0, -1 0, -1 -1, 0 0)),
+ *    POLYGON ((3 3, 4 4, 3 4, 3 3))
+ * ]
+ * Offset 0 2 3
+ * Offset 0 4 8 12
+ * Offset 0 2 4 6 8 10 12 14 16 18 20 22 24
+ * Child  0 0 1 1 0 1 0 0 0 0 -1 0 -1 -1 0 0 3 3 4 4 3 4 3 3
+ *
+ * @tparam T Coordinate value type
+ * @param polygon_offsets Offset to the starting position for each polygon
+ * @param ring_offsets Offset to the starting position for each ring
+ * @param point_coords Interleaved x-y coordinates of the points
+ * @param stream The CUDA stream on which to perform computations
+ *
+ * @return A cudf LIST column with polygon data
+ */
+template <typename T>
+std::pair<collection_type_id, std::unique_ptr<cudf::column>> make_polygon_column(
+  std::initializer_list<cudf::size_type>&& polygon_offsets,
+  std::initializer_list<cudf::size_type>&& ring_offsets,
+  std::initializer_list<T> polygon_coords,
+  rmm::cuda_stream_view stream)
+{
+  auto num_points = polygon_coords.size() / 2;
+  return {
+    collection_type_id::SINGLE,
+    make_non_nullable_lists_column<T>(
+      polygon_offsets,
+      make_non_nullable_lists_column<T>(
+        ring_offsets,
+        make_non_nullable_lists_column<T>(coords_offsets(num_points, stream), polygon_coords)))};
+}
+
+/**
+ * @brief helper function to make a polygon column
+ *
+ * A multilinestring column has cudf type LIST<LIST<LIST<FLOAT | DOUBLE>>>
```

Comment:
```suggestion
 * A polygon column has cudf type LIST<LIST<LIST<LIST<FLOAT | DOUBLE>>>>
```

Also I think that the rest of this doc needs to be updated reflecting the above suggestion.

Reply:
Only this cudf type is incorrect, the rest seems ok. I take this rule of thumb: multipoint and single-linestring has the same layout. Multilinestring and single-polygon has the same layout.